src/main/java/org/springframework/integration/kafka/core/DefaultConnection.java

```diff
@@ -0,0 +1,312 @@
+	}
+
+	/**
+	 * The broker address for this consumer
+	 *
+	 * @return broker address
+	 */
+	@Override
+	public BrokerAddress getBrokerAddress() {
+		return brokerAddress;
+	}
+
+	@Override
+	public void close() {
+		this.simpleConsumer.close();
+	}
+
+	/**
+	 * Fetches data from Kafka.
+	 *
+	 * @return a combination of messages and errors, depending on whether the invocation was successful or not
+	 * @throws ConsumerException
+	 */
+	@Override
+	public Result<KafkaMessageBatch> fetch(FetchRequest... requests) throws ConsumerException {
+		FetchRequestBuilder fetchRequestBuilder = new FetchRequestBuilder();
+		for (FetchRequest request : requests) {
+			Partition partition = request.getPartition();
+			long offset = request.getOffset();
+			int maxSize = request.getMaxSizeInBytes();
+			fetchRequestBuilder.addFetch(partition.getTopic(), partition.getId(), offset, maxSize);
+		}
+		FetchResponse fetchResponse;
+		try {
+			fetchResponse = this.simpleConsumer.fetch(fetchRequestBuilder.build());
+		}
+		catch (Exception e) {
+			throw new ConsumerException(e);
+		}
+		ResultBuilder<KafkaMessageBatch> resultBuilder = new ResultBuilder<KafkaMessageBatch>();
+		for (final FetchRequest request : requests) {
+			Partition partition = request.getPartition();
+			if (log.isDebugEnabled()) {
+				log.debug("Reading from " + partition + "@" + request.getOffset());
+			}
+			short errorCode = fetchResponse.errorCode(partition.getTopic(), partition.getId());
+			if (ErrorMapping.NoError() == errorCode) {
+				ByteBufferMessageSet messageSet = fetchResponse.messageSet(partition.getTopic(), partition.getId());
+				List<KafkaMessage> kafkaMessages = collect(messageSet, new ConvertToKafkaMessageFunction(request)).toList();
+				long highWatermark = fetchResponse.highWatermark(partition.getTopic(), partition.getId());
+				resultBuilder.add(partition).withResult(new KafkaMessageBatch(partition, kafkaMessages, highWatermark));
+			}
+			else {
+				resultBuilder.add(partition).withError(errorCode);
+			}
+		}
+		return resultBuilder.build();
+	}
+
+	@Override
+	public Result<Long> fetchStoredOffsetsForConsumer(String consumerId, Partition... partitions) throws
+			ConsumerException {
+		FastList<TopicAndPartition> topicsAndPartitions = FastList.newList(Arrays.asList(partitions))
+				.collect(new ConvertToTopicAndPartitionFunction());
+		OffsetFetchRequest offsetFetchRequest = new OffsetFetchRequest(consumerId, topicsAndPartitions,
+				kafka.api.OffsetFetchRequest.CurrentVersion(), createCorrelationId(), simpleConsumer.clientId());
+		OffsetFetchResponse offsetFetchResponse = null;
+		try {
+			offsetFetchResponse = simpleConsumer.fetchOffsets(offsetFetchRequest);
+		}
+		catch (Exception e) {
+			throw new ConsumerException(e);
+		}
+		ResultBuilder<Long> resultBuilder = new ResultBuilder<Long>();
+		for (Partition partition : partitions) {
+			OffsetMetadataAndError offsetMetadataAndError = offsetFetchResponse.offsets().get(partition);
```

Comment:
Yes, good catch. That's an unused/untested method, unfortunately. We need a test for it.